oracle/src/events/processCollectedSignatures/index.js

```diff
@@ -46,6 +47,34 @@ function processCollectedSignaturesBuilder(config) {
         logger.info(`Processing CollectedSignatures ${colSignature.transactionHash}`)
         const message = await homeBridge.methods.message(messageHash).call()
 
+        if (config.accessLists) {
+          const parsedMessage = parseMessage(message)
+          const recipient = parsedMessage.recipient.toLowerCase()
+          const originalTxHash = parsedMessage.txHash
+          const allowanceList = await readAccessListFile(config.accessLists.allowanceList)
+          const blockList = await readAccessListFile(config.accessLists.blockList)
+
+          if (allowanceList.length > 0) {
+            const sender = (await web3Home.eth.getTransaction(originalTxHash)).from.toLowerCase()
+            if (allowanceList.indexOf(recipient) === -1 && allowanceList.indexOf(sender) === -1) {
+              logger.info(
+                'Validator skips a transaction. Neither sender nor recipient addresses are in the allowance list.',
+                { sender, recipient }
+              )
+              return
+            }
+          } else if (blockList.length > 0) {
```

Comment:
Let's change the logic: check the block list first and if it is empty then the allowance list should be  handled. The reason is that the block list is going to be used more often.

Reply:
Sure, addressed in the last commit.